### PR TITLE
Ruby: Add some missing Rails sinks

### PR DESCRIPTION
**Pull request by hmac**
- Ruby: Add reorder as a SQL sink
- Ruby: Add count_by_sql as SQL sink
- Ruby: Taint flow through ActionController params
- Ruby: ActiveRecord::Connection.execute SQL sink

### CODE REVIEW

File: ruby/ql/lib/codeql/ruby/frameworks/ActiveRecord.qll [hunk 1]
```diff
@@ -208,11 +213,22 @@ class ActiveRecordSqlExecutionRange extends SqlExecution::Range {
     exists(PotentiallyUnsafeSqlExecutingMethodCall mc |
       this.asExpr().getNode() = mc.getSqlFragmentSinkArgument()
     )
+    or
+    exists(DataFlow::CallNode executeCall |
+      executeCall.getReceiver() = activeRecordConnectionInstance() and
+      executeCall.getMethodName() = "execute" and
+      this = executeCall.getArgument(0) and
+      unsafeSqlExpr(this.asExpr().getExpr())
```
Comment (asgerf) [inline ruby/ql/lib/codeql/ruby/frameworks/ActiveRecord.qll:221]:
```suggestion
    this = activeRecordConnectionInstance().getAMethodCall("execute").getArgument(0) and
    unsafeSqlExpr(this.asExpr().getExpr())
```

We get more benefit from global flow if we stay at the level of API nodes as long as possible before lowering to data-flow nodes.

File: ruby/ql/lib/codeql/ruby/frameworks/ActiveRecord.qll [hunk 2]
```diff
@@ -208,11 +213,22 @@ class ActiveRecordSqlExecutionRange extends SqlExecution::Range {
...
+    )
   }
 
   override DataFlow::Node getSql() { result = this }
 }
 
+private DataFlow::Node activeRecordConnectionInstance() {
+  result = activeRecordClassApiNode().getAMethodCall("connection")
+}
+
```
Comment (asgerf) [inline ruby/ql/lib/codeql/ruby/frameworks/ActiveRecord.qll:231]:
```suggestion
private API::Node activeRecordConnectionInstance() {
  result = activeRecordClassApiNode().getReturn("connection")
}
```

**Review by asgerf — COMMENTED** [review]
A quick comment about the use of API graphs, but otherwise LGTM (but maybe get a second pair of eyes on the new sinks; I haven't looked at Ruby in a while)

**Review by alexrford — DISMISSED** [review]
LGTM aside from Asger's suggestions.

**Review by asgerf — APPROVED** [review]

**Pull request merged**